### ISSUE TITLE
Printing as ZIP results in malformed ZIP

### ISSUE THREAD

**Issue by sevmonster** [clonemeagain/attachment_preview#49]
For example, `http://osticket/scp/tickets.php?id=1&a=zip` results in these warnings:
```
PHP Warning:  file_get_contents(): Unable to find the wrapper "phar" - did you forget to enable it when you configured PHP? in phar://osticket/include/plugins/attachment_preview.phar/class.AttachmentPreviewPlugin.php on line 169
PHP Warning:  file_get_contents(phar://osticket/include/plugins/attachment_preview.phar/stylesheet.css): failed to open stream: No such file or directory in phar://osticket/include/plugins/attachment_preview.phar/class.AttachmentPreviewPlugin.php on line 169
PHP Warning:  file_get_contents(): Unable to find the wrapper "phar" - did you forget to enable it when you configured PHP? in phar://osticket/include/plugins/attachment_preview.phar/class.AttachmentPreviewPlugin.php on line 186
PHP Warning:  file_get_contents(phar://osticket/include/plugins/attachment_preview.phar/script.js): failed to open stream: No such file or directory in phar://osticket/include/plugins/attachment_preview.phar/class.AttachmentPreviewPlugin.php on line 186
```
And ignoring those warnings spits out the following malformed ZIP file:
```
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
<html><body><p>PK</p><script></script>
</body></html>
```
I assume the NUL in the ZIP is causing the rest of the file to not display in the `<p>` tag.

![image](https://user-images.githubusercontent.com/16788697/96932122-fe9e7180-1483-11eb-908e-46d10d639e23.png)

**Comment by clonemeagain** [comment]
Printing as zip? I'm guessing we need to update the is ticket algorithm

**Comment by sevmonster** [comment]
Yes; don't know if you've seen it or when it was added, but this is what the current Print option looks like for a ticket in the SCP:
![image](https://user-images.githubusercontent.com/16788697/96934514-efb9be00-1487-11eb-9f7f-4088e8939dda.png)
The bottom two choices provide a ZIP file with attachments included.

**Comment by crazy-mc** [comment]
Hi are there any updates to this issue? My team love this plugin but every time they have to create a zip i must deactivate it and reactivate when it's done. Next time we want to update to 1.18 but i see there are also problems with compatibility

**Comment by qyak** [comment]
Not sure if any one needs this, but here is a fix (I'm not liable if you break something):

1. Make note of your plugin settings, and then uninstall plugin.
2. Re-Download or move attachments.phar to a folder outside of plugins
3. Run this: mkdir attachments_source
4. Then: php -r '$phar = new Phar("attachments.phar"); $phar->extractTo("attachments_source");'
5. cd attachments_source
6. Open class.AttachmentPreviewPlugin.php in your favorite text editor.
7. Search for this exact string: elseif (strpos($url, 'a=edit') || strpos($url, 'a=print'))
8. Modify it to look like this: elseif (strpos($url, 'a=edit') || strpos($url, 'a=print') || strpos($url, 'a=zip'))
9.  Save and exit.
10. cd ..
11. Build the .phar (build_attachment.phar attached).
12. Copy or move your new attachments.phar to your plugins folder.
13. Reinstall the plugin and reconfigure your settings.
14. Exporting should work now.

[build_attachement.php](https://github.com/user-attachments/files/25140818/build_attachement.php)

**Comment by JR-2026** [comment]
[attachments_patched.zip](https://github.com/user-attachments/files/28544840/attachments_patched.zip)
